src/revocation.rs

```diff
@@ -0,0 +1,131 @@
+#[macro_use]
+use log::*;
+use pretty_env_logger;
+
+use crate::common::config_get;
+use crate::crypto;
+use crate::error;
+use crate::error::{Error, Result};
+use crate::secure_mount;
+
+use std::path::Path;
+
+use serde_json::Value;
+
+/// Handles revocation messages via 0mq
+/// See:
+/// - URL: https://github.com/keylime/keylime/blob/master/keylime/revocation_notifier.py
+///   Function: await_notifications
+pub(crate) async fn run_revocation_service() -> Result<()> {
+    let mount = secure_mount::mount()?;
+    let revocation_cert_path =
+        format!("{}/unzipped/RevocationNotifier-cert.crt", mount);
+
+    // Connect to the service via 0mq
+    let context = zmq::Context::new();
+    let mysock = context.socket(zmq::SUB)?;
+
+    mysock.set_subscribe(b"")?;
+
+    let revocation_ip =
+        config_get("/etc/keylime.conf", "general", "receive_revocation_ip")?;
+    let revocation_port = config_get(
+        "/etc/keylime.conf",
+        "general",
+        "receive_revocation_port",
+    )?;
+    let endpoint = format!("{}:{}", revocation_ip, revocation_port);
+
+    mysock.connect(endpoint.as_str())?;
+
+    // Unlike the python agent we do not attempt lazy loading. We either
+    // have the certificate, or we don't. If we don't have a key or can't load
+    // the key we return a Configuration error as the service will not work.
+    let cert_key = if Path::new(&revocation_cert_path).exists() {
+        info!(
+            "Loading the revocation certificate from {}",
+            revocation_cert_path
+        );
+        match crypto::rsa_import_pubkey(revocation_cert_path.clone()) {
```

Comment:
If `rsa_import_pubkey` had accepted an `&str`, you wouldn't have needed to `.clone()` the path here. That's exactly why that is more idiomatic 😄

Reply:
Makes sense, thanks! I've updated the others but I'll leave this one be for now since it's in `rsa_import_pubkey`.